### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by AlexDrift** [pawn-lang/compiler#515]
EXAMPLE DEMO
new test[MAX_PLAYERS];
public OnPlayerClickPlayer(playerid, clickedplayerid, source)
{	
	test[playerid] = clickedplayerid;
	return 1;
}

Demo example
Ban(test[playerid]) ;
new test2[MAX_PLAYERS];
test2[test[playerid]]; and if(............) 
Here are examples of when I use it, and what doesn't happen.

**Comment by AlexDrift** [comment]
On this complained to my colleague, he, too, is used by your compiler. Sorry, I'm writing from my mobile

**Comment by Y-Less** [comment]
What?

**Comment by AlexDrift** [comment]
When I use this in your compiler application it doesn't work on the server

**Comment by AlexDrift** [comment]
3.10.10

**Comment by AlexDrift** [comment]
I write the ID of the player I clicked in the Tab to the array. After I try to issue a ban or use an array in the array on it through the if check. I compile and it doesn't work on the server, but the player ID is written to the array correctly and is not cleared.

**Comment by Y-Less** [comment]
Yes.  We need a working example of code.  Code that works in the old compiler but doesn't work on this compiler.

**Comment by AlexDrift** [comment]
OK, I'll give you the code later

**Comment by YashasSamaga** [comment]
@AlexDrift any news?

**Comment by skuzzis** [comment]
> EXAMPLE DEMO
> new test[MAX_PLAYERS];
> public OnPlayerClickPlayer(playerid, clickedplayerid, source)
> {
> test[playerid] = clickedplayerid;
> return 1;
> }
> 
> Demo example
> Ban(test[playerid]) ;
> new test2[MAX_PLAYERS];
> test2[test[playerid]]; and if(............)
> Here are examples of when I use it, and what doesn't happen.

this is not a clearly example, you could put the full code on pastebin or somewhere

**Comment by AlexDrift** [comment]
else if(dialogid == TEST_DIALOG+1)
	{
		if(response)
		{
			duelmoney = strval(inputtext);
		  	if(duelmoney < 10000 || duelmoney > 500000) return SendClientMessage(playerid, -1, !""MODE_NAME"Òû ââ¸ë íå âåðíóþ ñóììó ñòàâêè. Èñïîëüçóé ñóììó îò 10000$ äî 500000$");
		  	if(duelmoney > GetPlayerCash(playerid)) return SendClientMessage(playerid, -1, !""MODE_NAME"Òåáå íå õâàòàåò íà ñòàâêó. Óêàæè äðóãóþ ñóììó!");
		    if(duelmoney > GetPlayerCash(GetPVarInt(playerid,"ClickPlayer"))) return SendClientMessage(playerid, -1, !""MODE_NAME"Òâîåìó ïðîòèâíèêó íå õâàòàåò íà ñòàâêó. Óêàæè äðóãóþ ñóììó!");
		  	if(dragexits[GetPVarInt(playerid,"ClickPlayer")] == 1) return SendClientMessage(playerid, -1, !""MODE_NAME"Òâîé ïðîòèâíèê â äàííûé ìîìåíò â Äðàã äóýëå!");
		    if(dragexits[playerid] == 1) return SendClientMessage(playerid, -1, !""MODE_NAME"Âû â äàííûé ìîìåíò â Äðàã äóýëå!");
            ShowPlayerDialog(playerid, 3988, DIALOG_STYLE_LIST, !"Âûáåðèòå îðóæèå", !"Äèãë\nÄèãë è Øîò\nØîò\nÄèãë è Ñíàéïåðêà\nÌ4\nÀÊ-47", !"Âûáðàòü", !"Îòìåíà");
		}
	}

**Comment by AlexDrift** [comment]
the dialog ID is correct, I call the dialog, enter absolutely any number, and nothing happens. this is noticed with the transition to your compiler

**Comment by Y-Less** [comment]
That's still not a repro.

We need some code to copy and paste.  Put in a script and compile.  That, as is, won't compile.
